test/dial.spec.js

```diff
@@ -1,49 +1,70 @@
+/* eslint-env mocha */
+
 'use strict'
 
-/* eslint-env mocha */
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const pipe = require('it-pipe')
+const { collect } = require('streaming-iterables')
 
-const Client = require('..')
-const ID = require('peer-id')
-const IDJSON = [require('./id.json'), require('./id2.json')]
 const multiaddr = require('multiaddr')
-const SERVER_URL = multiaddr('/ip4/127.0.0.1/tcp/5892/ws/p2p-stardust')
-const pull = require('pull-stream/pull')
-const prom = (f) => new Promise((resolve, reject) => f((err, res) => err ? reject(err) : resolve(res)))
+const Stardust = require('../src')
+
+const { createPeer, SERVER_URL } = require('./utils')
+const mockUpgrader = {
+  upgradeInbound: maConn => maConn,
+  upgradeOutbound: maConn => maConn
+}
 
 describe('dial', () => {
   let clients = []
-  let conns = []
-  let ids
+  let listeners = []
 
   before(async () => {
-    ids = await Promise.all(IDJSON.map(id => prom(cb => ID.createFromJSON(id, cb))))
-  })
+    const peers = await createPeer({
+      number: 2
+    })
+
+    clients = peers.map((libp2p) => new Stardust({ upgrader: mockUpgrader, libp2p }))
+    listeners = clients.map(client => client.createListener(stream => pipe(stream, stream)))
 
-  it('connect all clients', async () => {
-    clients = ids.map(id => new Client({ id }))
-    conns = clients.map(client => client.createListener(conn => pull(conn, conn)))
-    await Promise.all(conns.map(conn => conn._listen(SERVER_URL)))
-    clients.forEach(c => (c.addr = multiaddr('/ipfs/' + c.id.toB58String())))
+    // start discovery
+    clients.forEach((client) => client.discovery.start())
+
+    await Promise.all(listeners.map(listener => listener.listen(SERVER_URL)))
+    clients.forEach(c => (c.addr = multiaddr('/p2p/' + c.id.toB58String())))
   })
 
-  it('client1 should dial client2 over server1 and echo back', async () => {
-    const conn = await conns[0]._dial(clients[1].addr)
-    const res = await prom(cb =>
-      pull(
-        pull.values(['hello']),
-        conn,
-        pull.collect(cb)
-      )
+  it('dial on IPv4, check promise', async function () {
```

Comment:
Test names should be clearer about what's being tested
```suggestion
  it('dial on IPv4 should return a valid connection', async function () {
```